projects/ontimize-web-ngx/src/lib/components/o-service-base-component.class.ts

```diff
@@ -354,12 +361,25 @@ export class OServiceBaseComponent implements ILocalStorageComponent, OnChanges
     }
   }
 
+  public getParentKeysFromContext(parentKeys: object, context: any) {
+    let result = {};
+    switch (true) {
+      case context instanceof OExpandableContainerComponent:
+        result = ServiceUtils.getParentKeysFromExpandableContainer(parentKeys, context);
+        break;
+      default:
+        result = ServiceUtils.getParentKeysFromForm(parentKeys, context);
+    }
+    return result;
```

Comment:
This `switch(true)` can be replaced by a `if else` statement or a ternary operator